### PR TITLE
Update @types/ember__routing RouterService#urlFor signature to match Ember typing

### PR DESCRIPTION
**Pull request by nlfurniss**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [ember.js code](https://github.com/emberjs/ember.js/blob/v4.3.0/packages/%40ember/-internals/routing/lib/services/router.ts#L243)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/ember__routing/router-service.d.ts [hunk 1]
```diff
@@ -268,6 +268,7 @@ export default class RouterService extends Service {
         modelsD: RouteModel,
         options?: { queryParams: object }
     ): string;
+    urlFor(routeName: string, ...args: RouteModel[] | [...RouteModel[], { queryParams: object }]): string;
```
Comment (chriskrycho) [inline types/ember__routing/router-service.d.ts:271]: So *theoretically*, this should subsume all the previous overloads, and you should be able to delete them and have the tests keep passing. Does that not actually prove out?

Comment (nlfurniss) [reply types/ember__routing/router-service.d.ts:271]: That was what I did first, but then wasn't sure whether that would necessitate a major bump so left them all as-is

Comment (chriskrycho) [reply types/ember__routing/router-service.d.ts:271]: Assuming the existing type tests pass we can remove them: the type tests exist precisely so we can make those kind of type level “refactoring” safely.

Comment (gabritto) [reply types/ember__routing/router-service.d.ts:271]: @nlfurniss I think you should remove the other signatures, and then we're good to go.

**Review by gabritto — CHANGES_REQUESTED** [review]

**Review by amcasey — APPROVED** [review]

**Comment by nlfurniss** [comment]
Ready to merge

**Pull request merged**